zcash_test_vectors/transaction.py

```diff
@@ -516,10 +517,16 @@ def to_bytes(self, version_bytes, nVersionGroupId, nConsensusBranchId):
             for desc in self.vSpendsSapling: # vSpendProofsSapling
                 ret += bytes(desc.proof)
             for desc in self.vSpendsSapling: # vSpendAuthSigsSapling
+                if hasSighashInfo:
```

Comment:
let's remove this param `hasSighashInfo` completly and replace with a runtime check for the class:
```python
if isinstance(self, (TransactionV6)):
```

Reply:
all checks (x2)

Reply:
or `tx.version_bytes() == NU7_TX_VERSION_BYTES:` to be unified with below